src/OpenCV/Contrib/Juicy.hs

```diff
@@ -0,0 +1,179 @@
+{-# language DataKinds, Rank2Types, TypeFamilies, ScopedTypeVariables, FlexibleContexts, ViewPatterns, ExistentialQuantification #-}
+-- | A thin JuicyPixels layer.
+module OpenCV.Contrib.Juicy (
+    -- * Types
+    Mat2D,
+    Filter ,
+    JC,
+    JD,
+    -- * Low level API
+    fromImage,
+    toImage,
+    -- * High level API
+    isoJuicy
+    )
+    where
+    
+import GHC.TypeLits (Nat,KnownNat)
+import Data.Proxy (Proxy (Proxy))
+
+import Foreign.Storable (Storable (..))
+import Foreign.Ptr (Ptr, plusPtr)
+import System.IO.Unsafe (unsafePerformIO)
+
+import Data.Word (Word8,Word16)
+import Data.Int (Int32)
+
+import Control.Monad (forM_)
+import Control.Monad.Primitive (PrimMonad)
+
+import Linear.V4 (V4)
+
+import OpenCV
+import OpenCV.Unsafe (unsafeRead, unsafeWrite)
+import Codec.Picture.Types 
+
+-- list of pointers at a given byte distance from a base one
+plusPtrS :: Ptr a -> Int -> [Ptr b]
+plusPtrS p n = map (plusPtr p) [0..n-1]
+
+-- multiple peek
+peekS p n = mapM peek (plusPtrS p n)
+
+-- multiple poke
+pokeS p n xs = sequence_ (zipWith poke (plusPtrS p n) xs)
+
+instance Storable (PixelRGB8) where
```

Comment:
We really shouldn't have orphaned instances in this library.

I propose creating our own `Storable` class:

```haskell
class StorablePixel pixel where
    peekPixel :: Ptr pixel -> IO pixel
    pokePixel :: Ptr pixel -> pixel -> IO ()
```